cpp/src/arrow/dataset/file_csv_test.cc

```diff
@@ -182,6 +183,18 @@ bar)");
     }
     ASSERT_EQ(rows, 5);
   }
+  {
+    SetSchema({field("custom_header", utf8())});
+    defaults->read_options.skip_rows = 0;
```

Comment:
0 is the default for `skip_rows`.  If you specify `column_names` then the CSV reader will automatically assume there is no header so you shouldn't have to manually set it to 0.  Maybe add a test case where `skip_rows` is non-zero to ensure you are passing it down correctly?

Reply:
This is resetting the value from above because the test is stateful - I'll come up with a better way to do this.